### PR TITLE
Remove down nodes from aten_sink

### LINKED ISSUE #7: aten_sink never garbage collects disconnected nodes.
aten_sink probably should maintain a set of node monitors for each node in it's monitored set and remove them from the set when a node is disconnected.

This will probably cause nodes that are disconnected suddenly never to be detected as down by `aten_detector` - additionally the `analyse` function will need to be updated to include the difference between the map of previous nodes and current as `down`

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #7 

@kjnilsson I have a couple TODOs for you to review to see if they make sense. Thanks!

[158857987]

### CODE REVIEW

**Comment by kjnilsson** [comment]
Thanks, this isn't quite what I had in mind. This handles the case where the failure detector detects a node as down. This is just an indicative result. All I wanted from this story was to handle the case when a node is disconnected and remove the state in the `aten_sink` in response to this. 

I suggest you maintain node monitors for each of the nodes in the `aten_sink` process and when you get a `node_down` for a node you remove it from the state and demonitor the node. Any node that never comes back to the cluster will at some point disconnect. If it reconnects the sink will begin to receive heartbeats again and will start collecting data again.

If [this call](https://github.com/rabbitmq/aten/blob/master/src/aten_detector.erl#L81) doesn't return any data from a node that was previously in `Prev` you will also need to take the difference there and emit `down` events _if_ their current state is `up`. Probably by handling the case when Curr here is `undefined` and `Prev` is not.

https://github.com/rabbitmq/aten/blob/master/src/aten_detector.erl#L117

Does this make sense? :)

**Comment by lukebakken** [comment]
Thanks for explaining, that makes sense. I thought, "why bother with a monitor when we know when a node is down" :smile:

File: src/aten_sink.erl [hunk 1]
```diff
@@ -78,6 +86,15 @@ code_change(_OldVsn, State, _Extra) ->
 %%% Internal functions
 %%%===================================================================
 
+maybe_monitor_node(Node, Monitors0) ->
+    IsMonitored = maps:get(Node, Monitors0, false),
+    case IsMonitored of
+        true -> Monitors0;
+        false ->
+            ct:pal("aten_sink node ~p monitoring node: ~p~n", [node(), Node]),
```
Comment (lukebakken) [inline src/aten_sink.erl]: This is weird @kjnilsson - I can see in the test output that the slave node is only monitored once, yet you can see in the output that the process receives hundreds of `nodedown` messages. I must be missing something. Also, two tests fail :-/

**Comment by lukebakken** [comment]
@kjnilsson all set

**Pull request merged**